src/app.js

```diff
@@ -0,0 +1,65 @@
+import NewsApiService from "./services/news.service";
+
+const API_KEY = "cd77286a82d24223b1320716435aacb6";
+
+export default class App {
+  constructor() {
+    this.service = new NewsApiService(API_KEY);
+  }
+
+  async init() {
+    this.sourceId = this.getQueryStringValue("sourceId");
+
+    let sources = await this.service.getSources();    
+    if (!this.sourceId.length && sources.length) {
+      this.sourceId = sources[0].id;
+    }
+
+    let articles = await this.service.getArticles(this.sourceId);
+
+    this.populateElement("sourcesList", sources, this.createSourceHtml);
+    this.populateElement("articlesList", articles, this.createArticleHtml);    
+  }
+
+  getQueryStringValue(parameter) {
+    let params = new URL(document.location).searchParams;
+    return params.get(parameter) || "";
+  }
+
+  populateElement(elementId, models, contentFactory) {
+    var element = document.getElementById(elementId);
+    element.innerHTML = "";
+
+    if (Array.isArray(models) && contentFactory) {
+      models.forEach(model => element.innerHTML += contentFactory.call(this, model));
```

Comment:
you don't need to use `call` and bind `this` as you don't use `this` in `createSourceHtml ` or `createArticleHtml `

Reply:
This is need to retrieve `this.sourceId` from current App obj.